src/server/qgsservicepythonloader.h

```diff
@@ -0,0 +1,73 @@
+/***************************************************************************
+                          qgsservicepythonloader.h
+
+  Define Loader for python service modules
+  -------------------
+  begin                : 2016-12-05
+  copyright            : (C) 2016 by David Marteau
+  email                : david dot marteau at 3liz dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSSERVICEPYTHONLOADER_H
+#define QGSSERVICEPYTHONLOADER_H
+
+class QgsServiceModule;
+class QgsServicePythonModuleEntry;
+class QgsServiceRegistry;
+
+#include "qgsserviceloader.h"
+#include <QHash>
+
+#include <memory>
+
+/**
+ * \ingroup server
+ * QgsServicePythonLoader
+ * Class defining the python service module loader for QGIS server services
+ *
+ * This class provides methods for loading and managing hook for native (C++) modules
+ *
+ */
+class SERVER_EXPORT QgsServicePythonLoader: public QgsServiceLoader
+{
+  public:
+    //! Constructor
+    QgsServicePythonLoader();
+
+    //! Destructor
+   ~QgsServicePythonLoader();
+
+    /**
+     * Lead all medules from path
+     * @param modulePath the path to look for module
+     * @parama registrar QgsServiceRegistry instance for registering services
+     */    
+    void loadModules( const QString& modulePath, QgsServiceRegistry& registrar ) override;
+
+    /**
+     * Unload all modules
+     */
+    void unloadModules() override;
+
+  private:
+    typedef QHash<QString, std::shared_ptr<QgsServicePythonModuleEntry> > ModuleTable;
+
+    /**
+     *  Unload medule hook 
+     */
+    void unloadModuleEntry( QgsServicePythonModuleEntry* entry );
+
+    //! Associative storage for module handles
+    ModuleTable mModules;
+};
+
+#endif
+
```

Comment:
This could be hard to implement for **not-core** services if we store this configuration in the project from the OWS option tab: how can QGIS Desktop running on a different machine know what Python services are available?
That's why I believe that core services (WMS, WFS, WCS) should be implemented in C++ and shipped with the QGIS Server package,  and leave Python services for plugins, user-overrides and custom services.